Functions/xASL_bids_BIDS2Legacy.m

```diff
@@ -72,15 +72,15 @@ function xASL_bids_BIDS2Legacy(pathStudy, bOverwrite, dataPar)
 end
 
 % Fills in important information in the dataPar if missing
-if ~isfield(dataPar,'x.subject_regexp')
+if ~isfield(dataPar.x,'subject_regexp')
```

Comment:
I think I know what you mean. We could make super sure that the script doesn't break here using:
```matlab
if ~isfield(dataPar,'x')
    dataPar.x = struct;
end
if ~isfield(dataPar.x,'subject_regexp')
    ...
end
```
Other than that I don't know what you're talking about... Didn't you write the part with the dataPar.x ?

Reply:
Haha no then perhaps Jan wrote it?
I'm fine here, I just mean that when somewhere (either reading the dataPar.json or reading an input struct) only have `x.*` rather than `dataPar.x.*`, that it manages this correctly.

E.g., if the `dataPar.json` normally is loaded into `x`, and has `x.subject_regexp`, and now suddenly has `x.dataPar.x.subject_regexp` or `x.x.subject_regexp`

But perhaps this is unnecessary precaution ;)

Reply:
Ahhh! I know what you mean and I think I understand why you think that this should be possible. Earlier we read json files as text files and we built the struct ourselves. Now that we use `spm_jsonread` whenever possible, we get code like this:
```matlab
json = spm_jsonread('file');
```
And if we have a structure like this within the json file...
```json
{"x": [{
	"name": "ExampleDataSet",
	}]}
```
... we always get a variable json with a field called x with a sub-field called name etc.

If you dont want this, we would have to change our json files to something like this:
```json
{
"name": "ExampleDataSet",
"subject_regexp": "...",
...
}
```

... but I don't think that's worth the effort just for code that's a bit shorter ;)